import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="4.18.0" severity="warning" />

Avoid passing asynchronous function as an argument where a synchronous function is expected.

### Example {#example}

**❌ Bad:**

```dart
void doSomethingWithCallback(VoidCallback function) {
  ...
  function();
  ...
}

void main() {
  doSomethingWithCallback(() async {
    await Future.delayed(Duration(seconds: 1));
    print('Hello World');
  });
}
```

**✅ Good:**

```dart
void doSomethingWithCallback(VoidCallback function) {
  ...
  function();
  ...
}

void main() {
  doSomethingWithCallback(() {
    print('Hello World');
  });
}
```
